File: C/implement_stack_using_queues.c
typedef struct {
    int data[100];
    int size;
    int top;
} MyStack;

/** Initialize your data structure here. */

MyStack* myStackCreate() {
    MyStack* myStack = malloc(sizeof(MyStack));
    myStack->size = 0;
    
    return myStack;
}

/** Push element x onto stack. */
void myStackPush(MyStack* obj, int x) {
    obj->data[obj->size] = x;
    obj->size ++;
    obj->top = x;
}

/** Removes the element on top of the stack and returns that element. */
int myStackPop(MyStack* obj) {
    if(obj->size == 0) return -1;
    int tmp = obj->top;
    obj->size --;
    if(obj->size == 0) return tmp;
    else obj->top = obj->data[obj->size-1];
    
    return tmp;
}

/** Get the top element. */
int myStackTop(MyStack* obj) {
    if(obj->size == 0) return -1;
    return obj->top;
}

/** Returns whether the stack is empty. */
bool myStackEmpty(MyStack* obj) {
    return obj->size == 0;
}

void myStackFree(MyStack* obj) {
    
}

/**
 * Your MyStack struct will be instantiated and called as such:
 * MyStack* obj = myStackCreate();
 * myStackPush(obj, x);
 
 * int param_2 = myStackPop(obj);
 
 * int param_3 = myStackTop(obj);
 
 * bool param_4 = myStackEmpty(obj);
 
 * myStackFree(obj);
*/

File: Python3/implement_stack_using_queues.py
from queue import Queue

class MyStack:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.q = Queue()
        self.tmp = Queue()
        
        self.size = 0

    def push(self, x: int) -> None:
        """
        Push element x onto stack.
        """
        self.q.put(x)
        self.size += 1

    def pop(self) -> int:
        """
        Removes the element on top of the stack and returns that element.
        """
        if self.q.empty(): return
        
        top_element = self.q.queue[self.size-1]
        for i in range(self.size-1):
            self.tmp.put(self.q.queue[i])
        self.q = self.tmp       
        self.size -= 1
        self.tmp = Queue()
        
        return top_element
        
    def top(self) -> int:
        """
        Get the top element.
        """
        if self.q.empty(): return
        return self.q.queue[self.size-1]
        
    def empty(self) -> bool:
        """
        Returns whether the stack is empty.
        """
        return self.q.empty()
        

# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()
